### PR TITLE
Support ETCD worker management for scheduler

### PR DESCRIPTION
**Pull request by jja725**
### What changes are proposed in this pull request?

Support ETCD worker management for scheduler

### Why are the changes needed?

na

### Does this PR introduce any user facing changes?
na

### CODE REVIEW

File: dora/core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java [hunk 1]
```diff
@@ -517,8 +522,24 @@ public Type getType() {
...
+    WorkerProvider workerProvider;
+    switch (Configuration.getEnum(PropertyKey.WORKER_MEMBERSHIP_MANAGER_TYPE,
+        MembershipType.class)) {
+      case STATIC:
+        workerProvider = new MembershipManagerWorkerProvider(
+            StaticMembershipManager.create(Configuration.global()), schedulerFsContext);
+        break;
+      case ETCD:
+        workerProvider = new MembershipManagerWorkerProvider(
+            EtcdMembershipManager.create(Configuration.global()), schedulerFsContext);
```
Comment (lucyge2022) [inline dora/core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java:534]: u could collapse these two and do 
new MembershipManagerWorkerProvider(
   MembershipManager.create()...
)

**Comment by lucyge2022** [comment]
LGTM! thanks for fixing the exception part!
But I would like to point out as Scheduler are using mActiveWorkers, change the var name to distinguish between ACTIVE(live) and REGISTERED workers. As I added the getLiveWorkerInfos in the WorkerProvider interface to distinguish the two so Scheduler can be aware of these two infos

File: dora/core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java (hunk 1 above)
Comment (jja725) [reply dora/core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java:534]: done

**Comment by jja725** [comment]
alluxio-bot, merge this please

**Pull request merged**